### PR TITLE
feat: add support for maxToolsUsePerRun in assistant builder

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Screenshot 2024-05-20 at 18 14 53

We want to allow configuring this value for multi actions agent (up to 8)

https://github.com/dust-tt/dust/issues/5153

## Risk

On critical path of AB, but well tested, including single-action assistant.

### CODE REVIEW

File: front/components/assistant/TryAssistant.tsx [hunk 1]
```diff
@@ -212,6 +212,7 @@ export function usePreviewAssistant({
   };
 
   const submit = useCallback(async () => {
+    console.log("submit", builderState);
```
Comment (PopDaph) [inline front/components/assistant/TryAssistant.tsx]: NIT: remove debug

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 2]
```diff
@@ -608,3 +623,57 @@ function ActionEditor({
...
+              </div>
+              <div className="w-full grow text-sm text-element-600">
+                up to {MAX_TOOLS_USE_PER_RUN_LIMIT}
+              </div>
+            </div>
+            <Input
+              value={maxToolsUsePerRun?.toString() ?? ""}
+              placeholder=""
+              name="maxToolsUsePerRun"
+              onChange={(v) => {
```
Comment (PopDaph) [inline front/components/assistant_builder/ActionsScreen.tsx:659]: No error display if I set "10"?

File: front/components/assistant_builder/ActionsScreen.tsx [hunk 3]
```diff
@@ -608,3 +623,57 @@ function ActionEditor({
...
+            <div className="flex flex-col items-start justify-start">
+              <div className="w-full grow text-sm font-bold text-element-800">
+                Max actions per run
+              </div>
+              <div className="w-full grow text-sm text-element-600">
+                up to {MAX_TOOLS_USE_PER_RUN_LIMIT}
+              </div>
+            </div>
+            <Input
+              value={maxToolsUsePerRun?.toString() ?? ""}
```
Comment (spolu) [inline front/components/assistant_builder/ActionsScreen.tsx:656]: We don't have type on Input?

**Review by spolu — APPROVED** [review]

File: front/components/assistant_builder/types.ts [hunk 4]
```diff
@@ -144,7 +146,8 @@ export function getDefaultAssistantState(): AssistantBuilderState {
       },
       temperature: 0.7,
     },
-  };
+    maxToolsUsePerRun: 3,
+  } satisfies AssistantBuilderState;
```
Comment (PopDaph) [inline front/components/assistant_builder/types.ts:150]: Why did you need to change that?

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 3 above)
Comment (spolu) [reply front/components/assistant_builder/ActionsScreen.tsx:656]: Would be nice to add. A text input for a number always looks broken-ish

**Review by PopDaph — APPROVED** [review]

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 3 above)
Comment (fontanierh) [reply front/components/assistant_builder/ActionsScreen.tsx:656]: Def agreed 👍

File: front/components/assistant_builder/ActionsScreen.tsx (hunk 2 above)
Comment (fontanierh) [reply front/components/assistant_builder/ActionsScreen.tsx:659]: You can't :p

File: front/components/assistant_builder/types.ts (hunk 4 above)
Comment (fontanierh) [reply front/components/assistant_builder/types.ts:150]: Because otherwise, the return value is just `AssistantBuilderState`, shadowing the fact that eg `maxToolsUsePerRun` is never null when coming out of this function

Comment (fontanierh) [reply front/components/assistant_builder/types.ts:150]: So here we really want typescript to infer the proper return type of the function BUT we also want that type to satisfy `AssistantBuilderState`
